### PR TITLE
layers: Refactor shader state

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Make all SHADER_MODULE_STATE data const.

This is related to both the upcoming locking refactorization as well as
general shader validation refactoring.

### CODE REVIEW

File: layers/shader_validation.cpp [hunk 1]
```diff
@@ -469,7 +469,7 @@ static std::set<uint32_t> TypeToDescriptorTypeSet(SHADER_MODULE_STATE const *mod
 
     switch (type.opcode()) {
         case spv::OpTypeStruct: {
-            for (auto insn : module->decoration_inst) {
+            for (auto insn : module->GetDecorationInstructions()) {
```
Comment (ziga-lunarg) [inline layers/shader_validation.cpp:472]: Should this be `const auto&`?

**Review by ziga-lunarg — APPROVED** [review]

File: layers/shader_validation.cpp (hunk 1 above)
Comment (ncesario-lunarg) [reply layers/shader_validation.cpp:472]: I'll take a look at the other lines modified in this change and add const accordingly to those as well.

File: layers/debug_printf.cpp [hunk 2]
```diff
@@ -460,8 +460,7 @@ std::vector<DPFSubstring> DebugPrintf::ParseFormatString(const std::string forma
 
 std::string DebugPrintf::FindFormatString(std::vector<unsigned int> pgm, uint32_t string_id) {
     std::string format_string;
-    SHADER_MODULE_STATE shader;
-    shader.words = pgm;
+    SHADER_MODULE_STATE shader(pgm);
```
Comment (ncesario-lunarg) [inline layers/debug_printf.cpp:463]: @Tony-LunarG I just realized that this differs from the previous behavior in that spirv-opt will run on the byte code if there are any "group decorations." If this is a problem, I can add an additional constructor to keep the pre-existing behavior.

Comment (Tony-LunarG) [reply layers/debug_printf.cpp:463]: The only reason the debug printf and GPU-AV code use SHADER_MODULE_STATE is to parse and search the spirv for something.  It's not ideal to run extra code analysis as part of the search, but it's worth the overhead to not have to find a lower overhead spirv parser (unless you know of one?).

**Pull request merged**